### PR TITLE
[Merged by Bors] - feat: test mqtt connector

### PR DESCRIPTION
**Pull request by EstebanBorai**
Test message consumption from MQTT using Cloud and Connectors

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,54 +63,3 @@ jobs:
       - name: Cargo clippy
         if: ${{ matrix.check == 'clippy' }}
         run: cargo clippy --all-targets -- -D warnings
-
```
Comment (sehz) [inline .github/workflows/ci.yml:60]: why is this removed?

File: .github/workflows/e2e_cloud.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+concurrency:
+  group: e2e-cloud-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - staging
```
Comment (sehz) [inline .github/workflows/e2e_cloud.yml:11]: e2e cloud test should be only run main branch and run nightly

**Review by sehz — CHANGES_REQUESTED** [review]

**Comment by galibey** [comment]
Since this connector already has integration tests written in Rust, I would propose choosing this approach for Cloud integration tests as well. It will give us more stability (no static sleeps) and better maintainability.

**Comment by EstebanBorai** [comment]
> Since this connector already has integration tests written in Rust, I would propose choosing this approach for Cloud integration tests as well. It will give us more stability (no static sleeps) and better maintainability. 

I really like the idea of using only Rust but the goal with these is to perform tests against CLI just as the user will do.

**Comment by galibey** [comment]
> > Since this connector already has integration tests written in Rust, I would propose choosing this approach for Cloud integration tests as well. It will give us more stability (no static sleeps) and better maintainability.
> 
> I really like the idea of using only Rust but the goal with these is to perform tests against CLI just as the user will do.

Yes, and this all can be done by Rust. For instance - https://github.com/infinyon/mqtt-connector/blob/cd1707211d0d2d7d54e6b35c58e481ab5c994673/integration-tests/src/main.rs#L312

File: .github/workflows/ci.yml (hunk 1 above)
Comment (EstebanBorai) [reply .github/workflows/ci.yml:60]: Actually this was a mistake sorry

File: .github/workflows/e2e_cloud.yml (hunk 2 above)
Comment (EstebanBorai) [reply .github/workflows/e2e_cloud.yml:12]: Fixed on https://github.com/infinyon/mqtt-connector/pull/23/commits/51b7bb563cc21c78937e3782c04ac99ef8aaf32f

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -58,53 +64,43 @@ jobs:
...
-  done:
-    name: Done
-    needs:
-      - check
-      - integration_tests
-    runs-on: ubuntu-latest
-    steps:
-      - name: Done
-        run: echo "Done!"
+ integration_tests:
```
Comment (sehz) [inline .github/workflows/ci.yml:67]: why is showing as replacement?

File: .github/workflows/e2e_cloud.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  group: e2e-cloud-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - staging
+  pull_request:
```
Comment (sehz) [inline .github/workflows/e2e_cloud.yml:13]: disable on PR.  can comment out

**Review by sehz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (EstebanBorai) [reply .github/workflows/ci.yml]: Is actually strange, given that we dont have any intentions to change this file Im reverting to current state.

File: .github/workflows/e2e_cloud.yml (hunk 4 above)
Comment (EstebanBorai) [reply .github/workflows/e2e_cloud.yml]: Done!

File: .github/workflows/e2e_cloud.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+concurrency:
+  group: e2e-cloud-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - staging
```
Comment (sehz) [inline .github/workflows/e2e_cloud.yml:12]: remove staging. this should be only run on main
and also run on schedule

**Review by sehz — CHANGES_REQUESTED** [review]

**Review by sehz — APPROVED** [review]
